Lib/test/test_grammar.py

```diff
@@ -65,6 +65,15 @@ def test_plain_integers(self):
         else:
             self.fail('Weird maxint value %r' % maxint)
 
+        if sys.py3kwarning:
+            with warnings.catch_warnings():
+                warnings.filterwarnings('error', category=Py3xWarning)
+                with self.assertRaises(Py3xWarning) as oct:
+                    compile('032', '<test string>', 'eval')
+                self.assertIn("octal literals are not supported in 3.x;\n" 
+                              "drop the leading 0",
```

Comment:
I don't think that dropping the leading `0` is the right fix: I *think* that the user has to translate their octal literal into another base entirely. This is an instance where customising the fix to to the user's input might be helpful (e.g. we might say "change 0o30 to 24").